### PR TITLE
🌱 Improve patch helper error handling

### PR DESCRIPTION
**Pull request by zhanggbj**
**What this PR does / why we need it**:
Improve patch helper error handling, as the patch package has improved to contain related context [here](https://github.com/kubernetes-sigs/cluster-api/pull/9946).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #2549

### CODE REVIEW

File: controllers/vspheredeploymentzone_controller.go [hunk 1]
```diff
@@ -287,17 +287,14 @@ func (r vsphereDeploymentZoneReconciler) reconcileDelete(ctx context.Context, de
...
-			obj.GetName())
+		return err
 	}
 
 	obj.SetOwnerReferences(ownerRefFunc())
 	if err := patchHelper.Patch(ctx, obj); err != nil {
-		return errors.Wrapf(err, "failed to patch object %s %s",
-			obj.GetObjectKind(),
-			obj.GetName())
+		return errors.Wrapf(err, "failed to update OwnerReferences")
```
Comment (chrischdi) [inline controllers/vspheredeploymentzone_controller.go:295]:
```suggestion
		return errors.Wrapf(err, "failed to patch OwnerReferences")
```

**Review by chrischdi — COMMENTED** [review]
Very nice cleanup, one small nit :-)

File: controllers/vspheredeploymentzone_controller.go (hunk 1 above)
Comment (chrischdi) [reply controllers/vspheredeploymentzone_controller.go:295]: Ignore this

**Comment by sbueringer** [comment]
Thx! Nice cleanup

/lgtm
/approve

**Pull request merged**